service/dap/server.go

```diff
@@ -735,9 +710,15 @@ func cleanExeName(name string) string {
 }
 
 func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
+	var args LaunchConfig
+	if err := mapToStruct(request.Arguments, &args); err != nil {
+		s.sendErrorResponse(request.Request,
+			FailedToLaunch, "Failed to launch", fmt.Sprintf("invalid debug configuration - %v", err))
+		return
+	}
 	// Validate launch request mode
-	mode, ok := request.Arguments["mode"]
-	if !ok || mode == "" {
+	mode := args.Mode
+	if mode == "" {
```

Comment:
What about defining a LaunchConfig constant with the defaults? This will both document what they are and allow you to unmarshall into a pre-populated struct. This way the defaults will stay if json doesn't specify alternatives.

Reply:
Good idea. Added LaunchMode and defined MarshalJSON. Same of AttachMode.
I think we can apply a similar trick for backend (but as a string type).

Still torn between 'int' and 'string' as their underlying type - Use of 'int' (and make "" and "debug" map to 0) helps avoiding this ""  (string zero value when mode is omitted) -> "debug" mapping code. On the other hand, this string in json <->int in the program mapping may not look obvious.